lib/dpl/provider/ops_works.rb

```diff
@@ -101,11 +101,24 @@ def create_deployment
         print "\n"
         if deployment[:status] == 'successful'
           log "Deployment successful."
+          return unless options[:update_app_on_success]
+          update_app
         else
           error "Deployment failed."
         end
       end
 
+      def update_app
+        update_config = {
+          app_id: option(:app_id),
+          app_source: {
+            revision: current_sha,
+          }
+        }
+        opsworks.update_app(update_config)
```

Comment:
I imagine that this could fail. It would be a good idea to check the result of this call and inform the user accordingly if it should fail.

Reply:
Yes, this once failed because the user did not have the correct permission:

```
Deploying application
creating deployment {"stack_id":"e4494","app_id":"f9e28","command":{"name":"deploy"},"comment":"Deploy build 806 via Travis CI","custom_json":"{\"deploy\":{\"bibcd\":{\"migrate\":false,\"scm\":{\"revision\":\"6ce403\"}}}}"}
Deployment created: 48b9abce
Deployment successful.
Deploying ........................................................
Already up to date!
HEAD detached at 6ce4031
nothing to commit, working tree clean
Dropped refs/stash@{0} (39a3e5a10165cfe324332610312435856d308b81)
Stopping Deploy, OpsWorks service error: arn:aws:iam::888666777:user/username is not authorized to perform: opsworks:UpdateApp
failed to deploy
```

So from that I assume that if an error happens on the opsworks side the it should be fine as the error message is displayed in the logs.

Do you think different?

Reply:
Ah, this is great.